internal/controller/components/dashboard/dashboard_support_test.go

```diff
@@ -0,0 +1,712 @@
+			g.Expect(err).ShouldNot(HaveOccurred())
+			g.Expect(variables).Should(Equal(tt.expected))
+		})
+	}
+}
+
+func TestComputeKustomizeVariableNoConsoleRoute(t *testing.T) {
+	ctx := t.Context()
+	g := NewWithT(t)
+
+	cli, err := fakeclient.New()
+	g.Expect(err).ShouldNot(HaveOccurred())
+
+	dsci := &dsciv1.DSCInitialization{
+		Spec: dsciv1.DSCInitializationSpec{
+			ApplicationsNamespace: testNamespace,
+		},
+	}
+
+	_, err = dashboard.ComputeKustomizeVariable(ctx, cli, cluster.OpenDataHub, &dsci.Spec)
+	g.Expect(err).Should(HaveOccurred())
+	g.Expect(err.Error()).Should(ContainSubstring("error getting console route URL"))
+}
+
+// TestComputeComponentNameBasic tests basic functionality and determinism with table-driven approach.
+func TestComputeComponentNameBasic(t *testing.T) {
+	tests := []struct {
+		name      string
+		callCount int
+	}{
+		{
+			name:      "SingleCall",
+			callCount: 1,
+		},
+		{
+			name:      "MultipleCalls",
+			callCount: 5,
+		},
+		{
+			name:      "Stability",
+			callCount: 100,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			// Test that the function returns valid component names
+			names := make([]string, tt.callCount)
+			for i := range tt.callCount {
+				names[i] = dashboard.ComputeComponentName()
+			}
+
+			// All calls should return non-empty names
+			for i, name := range names {
+				g.Expect(name).ShouldNot(BeEmpty(), "Call %d should return non-empty name", i)
+				g.Expect(name).Should(Or(
+					Equal(dashboard.LegacyComponentNameUpstream),
+					Equal(dashboard.LegacyComponentNameDownstream),
+				), "Call %d should return valid component name", i)
+				g.Expect(len(name)).Should(BeNumerically("<", 50), "Call %d should return reasonable length name", i)
+			}
+
+			// All calls should return identical results (determinism)
+			if tt.callCount > 1 {
+				for i := 1; i < len(names); i++ {
+					g.Expect(names[i]).Should(Equal(names[0]), "All calls should return identical results")
+				}
+			}
+		})
+	}
+}
+
+func TestInit(t *testing.T) {
+	g := NewWithT(t)
+
+	handler := &dashboard.ComponentHandler{}
+
+	// Test successful initialization for different platforms
+	platforms := []common.Platform{
+		cluster.OpenDataHub,
+		cluster.SelfManagedRhoai,
+		cluster.ManagedRhoai,
+	}
+
+	for _, platform := range platforms {
+		t.Run(string(platform), func(t *testing.T) {
+			err := handler.Init(platform)
+			g.Expect(err).ShouldNot(HaveOccurred())
+		})
+	}
+}
+
+func TestSectionTitleMapping(t *testing.T) {
+	g := NewWithT(t)
+
+	expectedMappings := map[common.Platform]string{
+		cluster.SelfManagedRhoai: "OpenShift Self Managed Services",
+		cluster.ManagedRhoai:     "OpenShift Managed Services",
+		cluster.OpenDataHub:      "OpenShift Open Data Hub",
+	}
+
+	for platform, expectedTitle := range expectedMappings {
+		g.Expect(dashboard.SectionTitle[platform]).Should(Equal(expectedTitle))
+	}
+}
```

Comment:
We are testing the map here, I'd avoid that

Reply:
why?

Reply:
Because we are testing the map key/value, creating another equal map. It is only a code duplication, not a real test assertion, and do not bring any value (if the original map is not correct, also this map would not be correct)